readthedocs_ext/external_version_warning.py

```diff
@@ -48,9 +48,7 @@ def process_external_version_warning_banner(app, doctree, fromdocname):
     ]
     prose = nodes.paragraph('', '', *children)
     warning_node = nodes.warning(prose, prose)
-
-    for document in doctree.traverse(nodes.document):
-        document.insert(0, warning_node)
+    doctree.insert(0, warning_node)
```

Comment:
I think the previous code adds the warning on _each_ document/page (`doctree.traverse(nodes.document)`) and the new code only adds the warning code in one document/page, doesn't it?